corehq/apps/sms/api.py

```diff
@@ -397,6 +399,9 @@ def process_incoming(msg, delay=True):
         msg.couch_recipient_doc_type = v.owner_doc_type
         msg.couch_recipient = v.owner_id
         msg.domain = v.domain
+        user = CommCareUser.get(v.owner_id)
+        if 'location_id' in user:
+            add_msg_tags(msg, metadata=MessageMetadata(location_id=user.location_id, workflow=msg.workflow))
         msg.save()
```

Comment:
We shouldn't be using CommCareUser.get() here because not all sms contacts are mobile workers. Instead this should look something like:

    contact = v.owner
    if isinstance(contact, CommCareUser) and hasattr(contact, 'location_id'):
        msg.location_id = contact.location_id

(There's no real need to use MessageMetadata here, that's mainly for setting fields in bulk, or more importantly, for passing the metadata to a send_sms* function)